docs/LANGUAGE.md

```diff
@@ -0,0 +1,388 @@
+# Opalog: OPA's Query Language
+
+OPA includes a policy engine which is purpose built for reasoning about information represented in structured documents such as JSON. Data stored in in the policy engine can be queried using OPA's native query language: Opalog.
+
+## What is Opalog?
+
+Opalog is a modernized version of [Datalog](https://en.wikipedia.org/wiki/Datalog) which is a well understood, decades old query language. Opalog extends Datalog to support structured document models such as JSON. Opalog queries are assertions on data stored in OPA. These queries can be used to define policies which enumerate violations that exist in the data stored in OPA.
+
+## Why use Opalog?
+
+Use Opalog for defining policy which is easy to read and write. 
+
+Opalog focuses on providing powerful support for referencing nested documents and ensuring that queries are correct and unambiguous.
+
+Opalog is declarative so policy authors can focus on what queries should return rather than how queries should be executed. These queries are simpler and more concise than the equivalent in an imperative language. Like other applications which support declarative query languages, OPA is able to optimize queries to improve performance, e.g, indexing, concurrent evaluation, reordering, etc.
+
+## Opalog Grammar
+
```

Comment:
It seems that this grammar is part of the introduction to the language, which would mean we should focus on making the grammar easy to understand.  10-12 lines is probably the largest anyone will try to scan through.  Maybe we could put the actual grammar at the end of the page and add a 'simplified grammar' for people to scan through.  Or maybe a short 'Overview' section here instead.